specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/WorkspacePurge.json

```diff
@@ -0,0 +1,216 @@
+      "description": "Describes the body of a purge request for an App Insights Workspace",
+      "required": [
+        "table",
+        "filters"
+      ],
+      "properties": {
+        "table": {
+          "type": "string",
+          "description": "Table from which to purge data."
+        },
+        "filters": {
+          "type": "array",
+          "description": "The set of columns and filters (queries) to run over them to purge the resulting data.",
+          "items": {
+            "$ref": "#/definitions/WorkspacePurgeBodyFilters"
+          }
+        }
+      }
+    },
+    "WorkspacePurgeBodyFilters": {
+      "description": "User-defined filters to return data which will be purged from the table.",
+      "properties": {
+        "column": {
+          "description": "The column of the table over which the given query should run",
+          "type": "string"
+        },
+        "operator": {
+          "description": "A query operator to evaluate over the provided column and value(s). Supported operators are ==, =~, in, in~, >, >=, <, <=, between, and have the same behavior as they would in a KQL query.",
+          "type": "string"
+        },
+        "value": {
+          "description": "the value for the operator to function over. This can be a number (e.g., > 100), a string (timestamp >= '2017-09-01') or array of values."
+        },
+        "key": {
+          "description": "When filtering over custom dimensions, this key will be used as the name of the custom dimension.",
+          "type": "string"
+        }
+      }
+    },
+    "WorkspacePurgeResponse": {
+      "description": "Response containing operationId for a specific purge action.",
+      "properties": {
+        "operationId": {
+          "description": "Id to use when querying for status for a particular purge operation.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "operationId"
+      ]
+    },
+    "WorkspacePurgeStatusResponse": {
+      "description": "Response containing status for a specific purge operation.",
+      "properties": {
+        "status": {
+          "description": "Status of the operation represented by the requested Id.",
+          "type": "string",
+          "enum": [
+            "pending",
+            "completed"
+          ],
+          "x-ms-enum": {
+            "name": "PurgeState",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "status"
+      ]
+    }
+  },
+  "parameters": {
+    "WorkspacePurgeParameter": {
+      "name": "body",
+      "in": "body",
+      "description": "Describes the body of a request to purge data in a single table of an Log Analytics Workspace",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/WorkspacePurgeBody"
+      },
+      "x-ms-parameter-location": "client"
+    },
+    "WorkspacePurgeIdParameter": {
+      "name": "purgeId",
+      "in": "path",
+      "required": true,
```

Comment:
why is it required.

Reply:
because its part of the path when doing GET /operations/..